language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/AddToConfigTomlCodeAction.java

```diff
@@ -0,0 +1,291 @@
+import org.ballerinalang.langserver.commons.codeaction.spi.RangeBasedPositionDetails;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.CodeActionKind;
+import org.eclipse.lsp4j.Command;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.TextEdit;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Code Action to undefined configurables into Config.toml file.
+ *
+ * @since 2201.10.0
+ */
+@JavaSPIService("org.ballerinalang.langserver.commons.codeaction.spi.LSCodeActionProvider")
+public class AddToConfigTomlCodeAction implements RangeBasedCodeActionProvider {
+
+    private static final String NAME = "Add to Config.toml";
+    private static final String CONFIG_TOML = "Config.toml";
+
+    @Override
+    public List<SyntaxKind> getSyntaxKinds() {
+        return List.of(SyntaxKind.MODULE_VAR_DECL);
+    }
+
+    @Override
+    public boolean validate(CodeActionContext context, RangeBasedPositionDetails positionDetails) {
+        return context.currentModule().isPresent()
+                && hasConfigurableQualifier((ModuleVariableDeclarationNode) positionDetails.matchedCodeActionNode());
+    }
+
+    @Override
+    public List<CodeAction> getCodeActions(CodeActionContext context, RangeBasedPositionDetails posDetails) {
+        String orgName;
+        try {
+            Project project = context.workspace().loadProject(context.filePath());
+            if (project.kind() == ProjectKind.SINGLE_FILE_PROJECT) {
+                return Collections.emptyList();
+            }
+            orgName = project.currentPackage().packageOrg().value();
+        } catch (Throwable e) {
+            return Collections.emptyList();
+        }
+
+        SemanticModel semanticModel = context.currentSemanticModel().get();
+        Types types = semanticModel.types();
+        TypeBuilder builder = types.builder();
+        TypeSymbol basicType = builder.UNION_TYPE.withMemberTypes(types.INT, types.STRING, types.BOOLEAN,
+                types.BYTE, types.FLOAT, types.DECIMAL, types.XML).build();
+        TypeSymbol anyDataOrJsonType = builder.UNION_TYPE.withMemberTypes(types.ANYDATA, types.JSON).build();
+
+        Optional<Symbol> confSymbol = semanticModel.symbol(posDetails.matchedCodeActionNode());
+        if (confSymbol.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        ConfigurableFinder.ModuleConfigDetails moduleConfigDetails = ConfigurableFinder.getModuleConfigDetails(context);
+        Map<String, ConfigurableFinder.ConfigVariable> configVarMap = moduleConfigDetails.configVariables();
+        VariableSymbol variableSymbol = (VariableSymbol) confSymbol.get();
+        ConfigurableFinder.ConfigVariable selectedConf = configVarMap.get(variableSymbol.getName().get());
```

Comment:
What happens if we can't find a `selectedConf` in `configVarMap`?

Reply:
We populate all the config variables into the `configVarMap` in the `getModuleConfigDetails` function. So the variable name always present in the configVarMap. We don't need to handle the error in this case.